### PR TITLE
refactor(root): a slightly more concise definition borrowed from Angular

### PR DESCRIPTION
**Pull request by benlesh**
**Description:**

**Related issue (if exists):**

### CODE REVIEW

**Comment by benlesh** [comment]
> so we're not throwing anymore if root is not resolvable?

I'm not sure there is a situation where root isn't resolvable, honestly. That was just some defensive coding. I'm indifferent, I'll leave it to the community.

**Comment by kwonoj** [comment]
I wouldn't expect this occurs in usual cases, but there were few exceptional cases patching globals around could be caught by thrown exception. So for me it's good to have, it allowed us to identify some unexpected behavior comes from it.

File: src/util/root.ts [hunk 1]
```diff
@@ -1,29 +1,26 @@
...
+// the global "global" var for now.
+
+const __window = typeof window !== 'undefined' && window;
+const __self = typeof self !== 'undefined' && typeof WorkerGlobalScope !== 'undefined' &&
+    self instanceof WorkerGlobalScope && self;
+const __global = typeof global !== 'undefined' && global;
+const _root: any = __window || __global || __self;
+
+if (!_root) {
+  throw new Error('No window, global or root reference found');
```
Comment (kwonoj) [inline src/util/root.ts]: I guess throw should be wrapped in IIFE as previous for closure compiler compatibility?

Comment (benlesh) [inline src/util/root.ts]: Yeah.. changed it to match what used to be there exactly.

**Pull request merged**